source/MaterialXRuntime/Private/Commands/PvtConnectionCmd.cpp

```diff
@@ -90,4 +95,33 @@ void PvtConnectionCmd::breakConnection(RtCommandResult& result)
     }
 }
 
+void PvtConnectionCmd::updateConnectionProperties(const RtOutput& /*src*/, const RtInput& /*dest*/, RtCommandResult& /*result*/) 
+{
+    // No-op
+}
+
+
+PvtCommandPtr PvtInterfaceConnectionCmd::create(const RtOutput& src, const RtInput& dest, ConnectionChange change)
+{
+    return std::make_shared<PvtInterfaceConnectionCmd>(src, dest, change);
+}
+
+void PvtInterfaceConnectionCmd::updateConnectionProperties(const RtOutput& src, const RtInput& dest, RtCommandResult& /*result*/)
+{
+    // Copy token information over
+    if (dest.isToken())
```

Comment:
@niklasharrysson , I made this port update a virtualization. As with discussed with Jonathan before this would not be required if we could set this at port creation time but we don't have enough information (only name + type is passed).

Reply:
I imagined PvtInterfaceConnectionCmd being implemented differently. This is the public API I prosed in our earlier discussion:

```
/// Connect a node input to the parent graph's input interface. The interface socket will be created
/// on the graph if it doesn't exists already.
void makeInterfaceConnection(const RtInput& nodeInput, const RString& graphInputName, RtCommandResult& result);

/// Connect a node output to the parent graph's output interface. The interface socket will be created
/// on the graph if it doesn't exists already.
void makeInterfaceConnection(const RtOutput& nodeOutput, const RtString& graphOutputName, RtCommandResult& result);
```

The socket on the nodegraph (here only given my name) would be created if it doesn't exists, and all information would be available internally to create it with the same flags as the node input it is to be connected to.

Reply:
I could do it this way, but doesn't this repeat existing connection code ?
Also,  the way things work currently is that ports are created first so it is assumed that it exsts but could add creation on demand but unsure what will occur as it could mismatch with what is created in VNN.

Reply:
It could still be derived from PvtConnectionCmd and use the same methods for making/breaking connections. Just that in addition it creates the interface input first if it doesn't exists, and with the correct flags set.

From a pure runtime API perspective it makes perfect sense to have such a command that encapsulates these operations into a single command, since it's a common task an API user would need. 

But I understand in LookdevX it's a problem of also having to follow the rules of how VNN operates. If the interface input is created first in VNN it might be better to just use the normal connection command and then work around it in LookdevX with low-level API (setting the flags after). I didn't consider this VNN issue when proposing we create a new command for this.

Reply:
Yes. I should have been prefaced that there is no way to create the correct port properties when it's dynamically created via VNN. There is still a "hack" outside the command code which copies over value data as well but I guess can only be done when the connection part is done (after creation). It lives outside the runtime command since it does it by setting it on the VNN port and then calling another runtime command which sets the value.

Maybe it's worth to try another go at finding some kind of proper callback from VNN to allow proper port creation.